components/eamxx/src/physics/mam/readfiles/vertical_remapper_mam4.cpp

```diff
@@ -0,0 +1,132 @@
+:VerticalRemapper(src_grid,
+                  tgt_grid,
+                  true,
+                  false) {
+  m_vremap_type=vremp_type;
+}
+
+void VerticalRemapperMAM4::remap_fwd_impl ()
+{
+  for (int i=0; i<m_num_fields; ++i) {
+    const auto& f_src    = m_src_fields[i];
+          auto& f_tgt    = m_tgt_fields[i];
+    apply_vertical_interpolation(f_src, f_tgt, m_src_pmid, m_tgt_pmid);
+
+  }
+}
+
+void VerticalRemapperMAM4::
+set_target_pressure (const Field& p)
+{
+  m_tgt_pmid=p;
+}
+
+void VerticalRemapperMAM4::
+set_source_pressure (const std::string& file_name )
+{
+  if (m_vremap_type == MAM4_ELEVATED_EMISSIONS) {
+    auto layout = m_src_grid->get_vertical_layout(false);
+    auto mbar = ekat::units::Units(100*ekat::units::Pa,"mbar");
+    Field altitude_int_src(FieldIdentifier("altitude_int_field",layout,mbar,m_src_grid->name()));
+    altitude_int_src.allocate_view();
+    scorpio::register_file(file_name,scorpio::FileMode::Read);
+    scorpio::read_var(file_name,"altitude_int",altitude_int_src.get_view<Real*,Host>().data());
+    altitude_int_src.sync_to_dev();
+    scorpio::release_file(file_name);
+    m_src_pmid=altitude_int_src;
+  }
+}
+
+void VerticalRemapperMAM4::
+apply_vertical_interpolation(const Field& f_src, const Field& f_tgt,
+                             const Field& p_src, const Field& p_tgt) const
+{
+  const auto p_tgt_c = p_tgt.get_view<const Real **>();
+  const auto datain = f_src.get_view<Real **>();
+  const auto dataout =  f_tgt.get_view<Real **>();
+
+  const auto& f_tgt_l = f_tgt.get_header().get_identifier().get_layout();
+  const int ncols = m_src_grid->get_num_local_dofs();
+  const int nlevs_tgt = f_tgt_l.dims().back();
+
+  const auto& f_src_l = f_src.get_header().get_identifier().get_layout();
+  const int levsiz = f_src_l.dims().back();
+
+  const auto policy =
+      ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(ncols, nlevs_tgt);
+  using Team = Kokkos::TeamPolicy<KT::ExeSpace>::member_type;
+  if (m_vremap_type== MAM4_PSRef) {
+
+    const int unit_factor_pin=1;
+    const auto p_src_c = p_src.get_view<const Real **>();
+
+    Kokkos::parallel_for(
+      "vert_interp", policy,
+      KOKKOS_LAMBDA(const Team &team) {
+        const int icol = team.league_rank();
+        const auto pin_at_icol     = ekat::subview(p_src_c, icol);
+        const auto pmid_at_icol    = ekat::subview(p_tgt_c, icol);
+        const auto datain_at_icol  = ekat::subview(datain, icol);
+        const auto dataout_at_icol = ekat::subview(dataout, icol);
+        mam4::vertical_interpolation::vert_interp(
+            team, levsiz, nlevs_tgt, pin_at_icol, pmid_at_icol, datain_at_icol,
+            dataout_at_icol, unit_factor_pin);
+          });
+  } else if (m_vremap_type== MAM4_ZONAL)
+  {
+    // unit conversion from mbar->pascals
+    const int unit_factor_pin=100;
+    const auto p_src_c = p_src.get_view<const Real *>();
+
+    Kokkos::parallel_for(
+      "vert_interp", policy,
```

Comment:
Thanks for adding names to these kernels. Can we add another string like `vert_interp_mam4_zonal`, so that they are more specific?